lib/lhm/connection.rb

```diff
@@ -8,36 +9,53 @@ class Connection < SimpleDelegator
     alias connection __getobj__
     alias connection= __setobj__
 
-    def initialize(connection:, default_log_prefix: nil, retry_options: {})
+    def initialize(connection:, default_log_prefix: nil, options: {}, retry_config: {})
       @default_log_prefix = default_log_prefix
-      @retry_options = retry_options || default_retry_config
+      @retry_options = retry_config || default_retry_config
       @sql_retry = Lhm::SqlRetry.new(
         connection,
-        retry_options,
+        retry_config,
+        options[:reconnect_with_consistent_host] || false
       )
 
       # Creates delegation for the ActiveRecord Connection
       super(connection)
     end
 
-    def execute(query, retry_options = {})
-      exec_with_retries(:execute, query, retry_options)
+    def execute(query, retriable: false, retry_options: {})
+      if retriable
+        exec_with_retries(:execute, query, retry_options)
+      else
+        exec(:execute, query)
+      end
     end
 
-    def update(query, retry_options = {})
-      exec_with_retries(:update, query, retry_options)
+    def update(query, retriable: false, retry_options: {})
+      if retriable
+        exec_with_retries(:update, query, retry_options)
+      else
+        exec(:update, query)
+      end
     end
 
-    def select_value(query, retry_options = {})
-      exec_with_retries(:select_value, query, retry_options)
+    def select_value(query, retriable: false, retry_options: {})
+      if retriable
+        exec_with_retries(:select_value, query, retry_options)
+      else
+        exec(:select_value, query)
+      end
     end
 
     private
 
+    def exec(method, sql)
+      connection.public_send(method, Lhm::ProxySQLHelper.tagged(sql))
```

Comment:
For the QueryLog, what I mean is if that is set, will the query be double tagged? Is that OK?

For `method_missing`, we don't need it now, but as we override more and more methods, we might consider using it. Let's defer that for the future tho as maybe at some point we can switch over to mysql2 instead of the AR connection.